code/obj/health_scanner.dm

```diff
@@ -27,84 +24,34 @@
 			src.id = new_id
 			boutput(user, "You change [src]'s ID to [new_id].")
 			src.find_partners()
-		else if (istype(W, /obj/item/device/analyzer/healthanalyzer_upgrade))
-			src.update_reagent_scan()
-			if (src.reagent_upgrade)
-				boutput(user, "<span style=\"color:red\">This system already has a reagent scan upgrade!</span>")
-				return
-			else
-				src.reagent_upgrade = 1
-				src.reagent_scan = 1
-				src.update_reagent_scan()
-				boutput(user, "<span style=\"color:blue\">Reagent scan upgrade installed.</span>")
-				playsound(src.loc ,"sound/items/Deconstruct.ogg", 80, 0)
-				qdel(W)
-				return
 		else
 			return ..()
 
 	proc/find_partners(var/in_range = 0)
 		return // dummy proc that the scanner and screen will define themselves
 
-	proc/accept_partner(var/obj/health_scanner/H)
-		if (!H)
-			return
-		if (locate(H) in src.partners)
-			return
-		src.partners += H
-
-	proc/update_reagent_scan()
-		if (!src.partners || !src.partners.len)
-			return
-		for (var/obj/health_scanner/myPartner in src.partners)
-			if (src.reagent_upgrade && !myPartner.reagent_upgrade)
-				myPartner.reagent_upgrade = 1
-			else if (myPartner.reagent_upgrade && !src.reagent_upgrade)
-				src.reagent_upgrade = 1
-			if (src.reagent_scan && !myPartner.reagent_scan)
-				myPartner.reagent_scan = 1
-			else if (myPartner.reagent_scan && !src.reagent_scan)
-				src.reagent_scan = 1
-
 /obj/health_scanner/wall
 	name = "health status screen"
 	desc = "A screen that shows health information recieved from connected floor scanners."
 	icon_state = "wallscan1"
+	var/list/partners // who do we know?
+	var/examine_range = 7 // from how far away can people examine the screen
```

Comment:
Is this supposed to be a hardcoded 7, or half of a screen height?

Reply:
I just put an arbitrary number but it'd be better if it depended on screen height. Should I just do something like `src.examine_range = (SQUARE_TILE_WIDTH - 1) / 2` under `New()`?

Reply:
i mean I'd just do it there, it's a define, but yes

Reply:
That's neat, ty.